tests/update/patch_sle.pm

```diff
@@ -12,65 +12,81 @@ use base "consoletest";
 use strict;
 use testapi;
 use utils;
-use version_utils qw(sle_version_at_least is_desktop_installed);
-use migration;
+use version_utils qw(is_sle is_desktop_installed);
 use registration;
 use qam;
 
-sub is_smt_or_module_tests {
-    return get_var('SCC_ADDONS', '') =~ /asmm|contm|hpcm|lgm|pcm|tcm|wsm|idu|ids/ || get_var('TEST', '') =~ /migration_offline_sle12sp\d_smt/;
+sub setup {
+    select_console 'root-console';
+
+    # Stop packagekitd
+    if (is_sle('12+')) {
+        pkcon_quit;
+    }
+    else {
+        assert_script_run "chmod 444 /usr/sbin/packagekitd";
+    }
+
+    # Change serial dev permissions
+    ensure_serialdev_permissions;
+
+    # Enable Y2DEBUG for error debugging
+    type_string "echo 'export Y2DEBUG=1' >> /etc/bash.bashrc.local\n";
+    script_run "source /etc/bash.bashrc.local";
```

Comment:
This function is almost same with `setup_migration` in lib/migration.pm, could you merge these two as a common one to reduce duplication?
Other than this LGTM

Reply:
Hmm, in that case, I have to bring "use migration" back ...

Reply:
And it looks a little strange to have "$self->setup_migration();" in this module renamed as "patch_sle.pm"

Reply:
Let me think more about it...

Reply:
Update: moved setup to migration and reused the same code to reduce the duplication
Verified with: http://openqa-apac1.suse.de/tests/846
@mitiao Please help check it again